ysh/val_ops.py

```diff
@@ -221,6 +221,24 @@ def FirstValue(self):
         return value.Str(self.strs[self.i])
 
 
+class RangeIterator(_ContainerIter):
+    """ for x in (m:n) { """
+
+    def __init__(self, val):
+        # type: (value.Range) -> None
+        _ContainerIter.__init__(self)
+        self.val = val
+        self.i = val.lower
```

Comment:
Hm `self.i` in the base class is supposed to be the index, not the value being iterated over

i.e. I think we should make `for i, x in (3:5)` work analogously like in Lists, and then this part should be a different variable